package/standalone_file_logger/standalone_file_logger/src/rotating_logger.cc

```diff
@@ -187,35 +187,65 @@ double RotatingLogger::get_time_passed()
 
 void RotatingLogger::frame_handler(const uint8_t *data, size_t size)
 {
-  if (!_dest_available) {
-    // check imediately on startup for path availability. Subsequently, check
-    // periodically
-    if (_session_start_time.time_since_epoch().count() != 0 && get_time_passed() < _poll_period) {
-      return;
+  // TODO: keep track of total data bytes: enforce a limit/warning in .h
+  std::vector<uint8_t> *frame = new std::vector<uint8_t> (data, data + size);
+
+  std::unique_lock<std::mutex> mlock(_mutex);
+  _queue.push_back( frame );
+  mlock.unlock();
```

Comment:
shouldn't need to unlock here - let the context take care of the lock's lifetime.

Reply:
fixed